### PR TITLE
[WIP]: Improve pod-template styling in the web console

### PR DESCRIPTION
**Pull request by rhamilto**
This is a first pass at improving the styling of the pod template in the web console.

Goals:

* better alignment with the new look and feel of the console and PatternFly without a complete overhaul (at least not yet)
* improved legibility

Actions:

* removed dotted border and rounded corners styling since we don't use these elsewhere
* removed max-width on pod template to remove negative space along right edge at large resolutions (note:  i expect debate on this one!  ;-) )
* increased spacing between elements and left justified all content
* softened the icon color to help differentiate it from other content per examples found in PatternFly [1][2]
* bolded the key in key value pairs so it's consistent with how we display keys elsewhere
* removed parenthesis around hash values and set the values in monospaced font to differentiate them without the need for the parens

[1]  aerogear-applist_2015-01-16

[2]  object-tablet_2015-02-02

Screenshots:

Note:  the overview page screenshots contain an unrelated bug with the "--o"

Before:
openshift web console 1

After:
openshift web console 2

Before:
openshift web console 3

After:
openshift web console 4

Before:
openshift web console 5

After:
openshift web console 6

What do you think, @jwforres, @spadgett, @sg00dwin, @benjaminapetersen, @ajacobs21e, @fabianofranz, et al.?

### CODE REVIEW

**Comment by spadgett** [comment]
@sg00dwin I like how it looks with the text left-aligned, but how does it look on mobile? Might need to stack at ~320px?

**Comment by benjaminapetersen** [comment]
Like the progress. Quick note -- the little magnifying glass on the left of the items on the overview screenshot seems to be cutoff.

**Comment by benjaminapetersen** [comment]
ah.

**Comment by rhamilto** [comment]
> I wonder if we can't just get rid of the border altogether on the browse pages. There can be a lot more content if you have volumes, resource limits, and health checks, which all appear inside the box. But those don't appear on the overview.

Just turning off the border isn't great.  What do y'all think?

openshift web console

**Comment by spadgett** [comment]
> Just turning off the border isn't great. What do y'all think?

Agree.

**Comment by jwforres** [comment]
what about just a heavy left border? something to visually associate it
still?

**Comment by rhamilto** [comment]
> what about just a heavy left border? something to visually associate it
still?

Better.  But I think I like the 1px border around the entire block better?

openshift web console

**Comment by spadgett** [comment]
We should also compare the styles when there's more content. There could be 10+ lines if you have volumes, resource limits, and health checks.

**Comment by benjaminapetersen** [comment]
Of the options so far, i like the heavy left border.   Do think it would be nice to further improve, though.

**Comment by rhamilto** [comment]
There seems to be some consensus around the left border, so I've implemented that change.  Additionally, I uppercased .component-label in all contexts (previously they were only uppercased on overview) and I [corrected the markup semantics on deployment pages](https://github.com/rhamilto/origin/commit/986b09d2dcba055199788edd1f054e712f8b11aa#diff-7ddde0cd2069455e87ecbc69b215a13bR112), which changes the display, but I think it works now that the border doesn't wrap the entire block.  Not included is @sg00dwin's feedback about left alignment.  That's proving difficult to get consistency across all the different contexts.  :/  Thoughts?

Before:
openshift web console 1

After:
openshift web console 2

openshift web console 2 1

Before:
openshift web console 3

After:
openshift web console 4

openshift web console 4 1

Before:
openshift web console 5

After:
openshift web console 6

openshift web console 6 1

Before:
openshift web console 7

After:
openshift web console 8

openshift web console 8 1

**Comment by benjaminapetersen** [comment]
Def an improvement. Noticed sometimes it is right below the "Pod Template" heading, sometimes its to the right (in non-mobile contexts).  Thoughts?  It appears that we are moving towards two column look (headings to left, content to right), maybe Im noticing the ones we haven't updated yet.

**Comment by spadgett** [comment]
I don't think it should appear in a column in the definition list. There's too much content, and we already have wrapping problems. Block element with consistent headers across the browse pages? I wonder if the "Pod template:" is even necessary.

@jwforres How do you feel about moving selectors and replicas into the config section for browse deployment, keeping only the pod template under "Template"?

It's definitely looking better.

**Comment by rhamilto** [comment]
> I don't think it should appear in a column in the definition list. There's too much content, and we already have wrapping problems. Block element with consistent headers across the browse pages? I wonder if the "Pod template:" is even necessary.

I wondered the same thing, @spadgett.  

@all, along the same lines, any insight why "Template" is missing and "Pod template" isn't bolded on https://cloud.githubusercontent.com/assets/895728/13746318/36f69ade-e9c9-11e5-881c-893ce9a434de.png

**Comment by ajacobs21e** [comment]
@rhamilto I agree with the left border line and left aligning the entire block. Thanks for working on this.

What does it look like on overview with the left border?

**Comment by rhamilto** [comment]
> What does it look like on overview with the left border?

Kind of funky, IMO.  I think [the 1px border on all sides](https://cloud.githubusercontent.com/assets/895728/13674403/5b97fe26-e6ab-11e5-9cb3-5d32f03f9b38.png) looks better in this context.  Thoughts?

openshift web console

**Comment by benjaminapetersen** [comment]
Hmm...fwiw, I may +1 the consistent look, actually.

**Comment by ajacobs21e** [comment]
@rhamilto also +1 to consistency. on overview I'd add more padding between the donut and template though

**Comment by spadgett** [comment]
I'm undecided on the new style on the overview. We do have compact and detailed pod templates. Detailed has more content and is used in the browse pages. We could use different styles for each.

How does it looks when the pod template has more than one container?

`oc create -f https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/pods/pod_with_two_containers.json`

**Comment by spadgett** [comment]
It looks like replicas is under "Template" to be consistent with the CLI, but that's changing.

https://github.com/openshift/origin/pull/7995#issuecomment-196784168

@jwforres I think we should reorganize some sections to be consistent with CLI and across browse pages.

**Comment by jwforres** [comment]
actually i suspect it was the other way around, pretty sure we chose to split it out under Template when we were moving things around to use the whitespace.  But I have no problem with re-organizing this stuff however it makes sense.

I actually don't mind the new pod template style on overview, reduces the boxiness, but agree with @spadgett that we need to see it with multiple containers because that was the original reason we added the box around it

**Comment by rhamilto** [comment]
> How does it looks when the pod template has more than one container?

1px border
screen shot 2016-03-15 at 9 55 52 am

3px border-left
screen shot 2016-03-15 at 9 56 20 am

**Comment by spadgett** [comment]
I like the box better, at least for overview :/  But I am OK with either.

**Comment by rhamilto** [comment]
> I'd add more padding between the donut and template though

Adds 10px.  That enough?

openshift web console
openshift web console 2

**Comment by benjaminapetersen** [comment]
1st looks good.  2nd... kinda squishy eh?

**Comment by jwforres** [comment]
well i think the second one is supposed to get fixed when we can change the
breakpoints (so they would be stacked), with the new bootstrap variable
coming in the next patternfly version

**Comment by spadgett** [comment]
We should be able to stack if we want without the bootstrap update if we're talking about the second picture here: https://github.com/openshift/origin/pull/7929#issuecomment-196842833. We're using the mobile/tablet flexbox attributes for column or row layout.

**Comment by spadgett** [comment]
@rhamilto How does it look if "Container: name" is above the div with the left border, container title left-aligned with the border? Wondering if that makes it more obvious when there's more than one container.

**Comment by rhamilto** [comment]
> @rhamilto How does it look if "Container: name" is above the div with the left border, container title left-aligned with the border? Wondering if that makes it more obvious when there's more than one container.

That could work. 

screen shot 2016-03-15 at 4 04 26 pm
screen shot 2016-03-15 at 4 05 51 pm

But I think it feels a bit more like an atomic whole if the border wraps all parts of the template.  Thoughts?

**Comment by spadgett** [comment]
Having the header outside separates the containers more clearly for me when there are two.

**Comment by jwforres** [comment]
i think i like the header better outside

**Comment by benjaminapetersen** [comment]
Works for me!

**Comment by rhamilto** [comment]
@sg00dwin @ajacobs21e, want to chime in with your votes on [heading location](https://github.com/openshift/origin/pull/7929#issuecomment-197000297)?

**Comment by sg00dwin** [comment]
+1 outside the border works better

**Comment by ajacobs21e** [comment]
a little late but yes agreed header outside of border. 

@jwforres @spadgett what about removing the label (e.g. "build") at mobile and just using the icon?

**Comment by jwforres** [comment]
@ajacobs21e the icons on the overview might be understandable without labels, but i would worry about some of the more advanced ones on the detailed pod template on the other pages, like mounts and resource constraints

**Comment by spadgett** [comment]
We could stack the donut and pod template sooner. Or collapse the sidebar at medium resolutions (eek).

**Comment by ajacobs21e** [comment]
@jwforres I meant on overview since it's so squished as is. The fields seem different enough that icons may be enough.

**Comment by rhamilto** [comment]
> @jwforres How do you feel about moving selectors and replicas into the config section for browse deployment, keeping only the pod template under "Template"?

So move lines [105-110](https://github.com/openshift/origin/blob/master/assets/app/views/browse/deployment-config.html#L105) to [which lines of Configuration](https://github.com/openshift/origin/blob/master/assets/app/views/browse/deployment-config.html#L84)?

screen shot 2016-03-22 at 9 50 39 am

Also, shouldn't [line 87](https://github.com/openshift/origin/blob/master/assets/app/views/browse/_deployment-details.html#L87) of deployment details be the same as [line 103](https://github.com/openshift/origin/blob/master/assets/app/views/browse/deployment-config.html#L103) of the deployment config so that we always display the pod template underneath the Template h3?

screen shot 2016-03-22 at 9 50 27 am

cc: @spadgett

**Comment by spadgett** [comment]
> So move lines 105-110 to which lines of Configuration?

I'd say make them the first two lines under configuration to (more or less) match the CLI:

https://github.com/openshift/origin/pull/7995#issuecomment-196784168

> Also, shouldn't line 87 of deployment details be the same as line 103 of the deployment config so that we always display the pod template underneath the Template h3

I think so.

**Comment by rhamilto** [comment]
Thanks, @spadgett.  Done.  PTAL, everyone.

screen shot 2016-03-22 at 4 42 08 pm
screen shot 2016-03-22 at 4 42 17 pm

File: assets/app/styles/_components.less [hunk 1]
```diff
@@ -160,25 +159,49 @@
...
-  margin-right: 10px;
+}
+
+.pod-template-block {
+  + .pod-template-block {
+    margin-top: 15px;
+  }
+  dd & {
+    margin-top: 2px;
+  }
```
Comment (spadgett) [inline assets/app/styles/_components.less]: Can we remove this rule? Are we showing the template inside a `dd` now?

File: assets/app/styles/_components.less [hunk 2]
```diff
@@ -266,7 +296,7 @@
 .pod-block {
   .overview-pods {
     margin-right: 10px;
-    min-width: 190px;
+    min-width: 200px;
```
Comment (spadgett) [inline assets/app/styles/_components.less]: #7678 removes the min-width entirely. Is the large size to give more spacing?

**Comment by rhamilto** [comment]
> #7678 removes the min-width entirely. Is the large size to give more spacing?

Yeah, per [@ajacobs21e's comment](https://github.com/openshift/origin/pull/7929#issuecomment-196619531).

**Comment by spadgett** [comment]
Can we use margins instead? The donut will no longer be fixed widths because the scaling controls will not always be visible.

File: assets/app/styles/_components.less [hunk 3]
```diff
@@ -4,7 +4,6 @@
 
 .components {
   border-left: 3px solid #ccc;
-  border-left-style: dotted;
```
Comment (spadgett) [inline assets/app/styles/_components.less]: This changes the style when a replication controller is not under a service. Was that intentional? It used to have a dotted line.

openshift_web_console

Now it doesn't:

openshift_web_console

**Comment by spadgett** [comment]
I definitely prefer the new styles.

File: assets/app/styles/_components.less (hunk 1 above)
Comment (rhamilto) [inline assets/app/styles/_components.less]: Yep.  Good catch.

File: assets/app/styles/_components.less (hunk 3 above)
Comment (rhamilto) [inline assets/app/styles/_components.less]: That's a bug I introduced.  Will fix.

File: assets/app/styles/_components.less (hunk 2 above)
Comment (rhamilto) [inline assets/app/styles/_components.less]: We can switch to margin, but that has the effect of pod-templates not aligning horizontally when the scaling controls are not present.  Is that what we want?

screen shot 2016-03-23 at 9 41 00 am

Comment (spadgett) [inline assets/app/styles/_components.less]: Hm. I think with the new styles, it's better to keep them left-aligned. I'll undo that change in my autoscaling pull request.

**Comment by rhamilto** [comment]
@spadgett, I pushed the fixes.

**Comment by spadgett** [comment]
Thanks @rhamilto. LGTM

**Comment by rhamilto** [comment]
Will merge via @jwforres' console_1.3 branch.  See https://github.com/jwforres/origin/pull/22

**Pull request closed without merging**